### ISSUE TITLE
Dependency conflict with Unity's own aab solution

### ISSUE THREAD

**Issue by NickBack** [google/play-unity-plugins#138]
We're using Unity's new auto split .aab solution which works great for us.

But we also want to use this plugin for in-App Review.

The problem is that the in-App Review adds playcore.aar which conflicts with com.google.android.play:core:1.10.0 that Unity adds when you tell it to use split .aab.

I'm guessing playcore.aar and com.google.android.play:core:1.10.0 are almost identical, but which one is a superset of the other?

In other words, which of these dependencies can I safely delete without it being an issue?

(also why isn't this handled by the External Dependency Manager?)

**Comment by jamcohen** [comment]
Hi @NickBack,

Thanks for calling this out. We're exploring ways to address this incompatibility and External Dependency Manager is one of the options we're looking into.

Right now, playcore.aar and com.google.android.play:core:1.10.0 are identical so you can safely remove either one.

**Comment by Hitomilras** [comment]
Oh thanks you...

**Comment by manwithsteelnerves** [comment]
For the time being, here is a post processor to remove the entry added by Unity (for PAD auto built-in support) if your project uses latest play core libraries.

Place the attached file in Editor folder.
[Here is the post on unity forum](https://forum.unity.com/threads/how-to-successfully-sdk30-and-gradle-7-x.1169648/#post-9446324).

```
#if UNITY_EDITOR && UNITY_ANDROID
using System.IO;
using UnityEditor.Android;
using System.Text;
using UnityEngine;
using System.Collections.Generic;
using System.Linq;

namespace VoxelBusters.EssentialKit.Editor.Build.Android
{
    public class PatchDuplicatePlayCoreLibrariesGradlePostProcessor : IPostGenerateGradleAndroidProject
    {
        private string playCoreOldDependencyEntry = "com.google.android.play:core:";

        public int callbackOrder { get { return int.MaxValue; } }

        public void OnPostGenerateGradleAndroidProject(string basePath)
        {
            string gradlePath = Path.Combine(basePath, "build.gradle");
            Debug.Log("Updating build.gradle at : " + gradlePath);

            string[] lines = File.ReadAllLines(gradlePath);
            StringBuilder builder = new StringBuilder();

            List<string> priorityDependencyEntries = GetPriorityDependencyEntries();
            bool hasPriorityEntries = false;

            foreach (string each in lines)
            {
                if (priorityDependencyEntries.Any(eachEntry => each.Contains(eachEntry)))
                {
                    hasPriorityEntries = true;
                }

                if(!(hasPriorityEntries && each.Contains(playCoreOldDependencyEntry))) //Considering old play core dependency will be added by unity at last
                {
                    builder.AppendLine(each);
                }
                else
                {
                    builder.AppendLine($"/*{each} => Patched by {this.GetType().Name}.cs file*/");
                }
            }
            File.WriteAllText(gradlePath, builder.ToString());
        }

        private List<string> GetPriorityDependencyEntries()
        {
            List<string> list = new List<string>();
            list.Add("com.google.android.play:core-common");
            list.Add("com.google.android.play.core-common-");
            list.Add("com.google.android.play:review");
            list.Add("com.google.android.play.review-");

            return list;
        }
    }
}
#endif
```

Cheers,
VB Team
